go-controller/pkg/controller/controller.go

```diff
@@ -201,7 +201,22 @@ func (c *controller[T]) onUpdate(oldObjInterface, newObjInterface interface{}) {
 }
 
 func (c *controller[T]) onDelete(objInterface interface{}) {
-	key, err := cache.MetaNamespaceKeyFunc(objInterface)
+	obj, ok := objInterface.(*T)
+	if !ok {
+		tombstone, ok := objInterface.(cache.DeletedFinalStateUnknown)
```

Comment:
I think you can use:
```
       // get the key for our object, which may have been deleted already
	if key, err = cache.DeletionHandlingMetaNamespaceKeyFunc(obj); err != nil {
		utilruntime.HandleError(blah blah err)
		return
	}
```
this straight? though it seems tiny bit different since it doesn't do the tombstone error reporting but it seems to be the client-go equivalent of what should be called for deletion?